### ISSUE TITLE
Sync bookmarks

### ISSUE THREAD

**Issue by bbinto** [mozilla-mobile/fenix#919]
As a user, I want to retrieve my bookmarks from my other devices, so I can retrieve previoulsy saved web content

### Why/User Benefit/User Problem
- allow users to retrieve their bookmarks saved on a different device
- allow users to delete bookmark and apply change on the server (change reflected on all other devices)

### What / Requirements
- ability to sync bookmarks from other devices via Firefox Account Sync

### Acceptance Criteria (how do I know when I’m done?)
- make clear to the user if bookmark was synced or not (what device)
- successfully sync bookmarks both ways

**Comment by grigoryk** [comment]
> make clear to the user if bookmark was synced or not (what device)

In our current Sync-connected products, we mostly just display a high level sync status on error. We could go a bit more granular here (e.g. history sync succeeded but bookmark sync failed), but this will require some in-depth UX thinking.

**Comment by colintheshots** [comment]
I feel like a granular error status might be too much information for the user. I suppose we could have some kind of synced/syncing/sync error indicator for each page in the Library if that's a thing, but I actually prefer to hide this kind of inside baseball. Most users only care on a very high level.

**Comment by grigoryk** [comment]
Yup. rfeeley will be the person to talk to about this. I'd vote for a single "sync status" to begin with.

**Comment by mheubusch** [comment]
@bbinto @colintheshots I have created a Meta for the Library #1140 and created a user story for syncing bookmarks and history to/from the Library #1155 so can we close this issue? I don't think we have a precedent for syncing only one type of thing or displaying an error status for one type of thing - I believe sync is all or nothing for the things a user wants to sync and for what the app can accept

**Comment by bbinto** [comment]
@mheubusch - #1155 might be now a duplicate of #289.

We do have this epic/meta: https://github.com/mozilla-mobile/fenix/issues/922
I'm removing the assigned epic for this issue and assign #1155 issue to the #922 epic, sounds good?

**Comment by mheubusch** [comment]
@ryanfeeley Looks like there is a question to you here that is blocking implementation.  Can you follow up?

**Comment by ryanfeeley** [comment]
> make clear to the user if bookmark was synced or not (what device)

We don't need this nor do we do this anywhere else. Sync just works. Yes, there may be occasional errors in syncing, but we don't present them to the user. In desktop there is about:sync-log where we direct users who report problems.

**Comment by bifleming** [comment]
@colintheshots looks like this is unblocked.

**Comment by sv-ohorvath** [comment]
Reopening so I can add blocking issues.

**Comment by bifleming** [comment]
Pulling from current sprint, so it can go out in the second beta release.

**Comment by kbrosnan** [comment]
Removing QA needed for now. When this re-lands please add it back.

**Comment by kbrosnan** [comment]
Need #2252 fixed to verify this.

**Comment by mheubusch** [comment]
Closing this - 2252 is  still open and work is tracked there.

**Comment by kbrosnan** [comment]
This needs to go through the correct process.

**Comment by grigoryk** [comment]
#2252 landed, will be in tomorrow's build.

**Comment by softvision-miralobontiu** [comment]
Verified that the bookmarks and history are synced on the latest build 1.0.1921 (#11440614), with Sony Xperia Z5 Premium (Android 7.1.1), Google Pixel (Android 9), and Huawei Honor 8 (Android 7).
